examples/vpc_service_control/README.md

```diff
@@ -0,0 +1,129 @@
+
+1. Create the on-prem project.
+    - In your favorite text editor, edit and save the following variables in `onprem_project/terraform.tfvars`:
+        - `project_id`: the unique ID for the on-prem project that is going to be created.  Use lowercase letters and a 6 digit random number only (no spaces).
+        - `service_account_file`: the name of the service account secret key file you downloaded.  Relative path is fine if you saved it in the `onprem_project` directory.
+        - `organization_id`: run `gcloud organizations list` to get this **number**.
+        - `billing_account_id`: the billing account that the project will charge to.
+        - `cloud_router_region`: it can be any region you like, although I recommend sticking with us-west1 because I haven't tested other regions.
+        - `shared_secret_string_for_vpn_connection`: any random text string can go here.
+        - Leave the rest blank.
+    - From a command line terminal:
+        - `cd onprem_project`
+        - `terraform plan`
+            - It should show a plan to create one project, two routes, and one firewall rule.  
+        - `terraform apply` and type "yes" when it prompts you to create the resources.  Might take a few minutes to create the project (should be less than 5 minutes).
+        - You will get an error message saying the Compute Engine API isn't enabled.  There will be a URL in the error message.  Copy and paste the URL into a browser, and then enable the Compute Engine API (takes a few minutes; be patient, let the spinner spin).  Once it's enabled, run `terraform apply` again - it should succeed.
+
+2. Create the project that will be protected by VPC Service Controls.
+    - Edit and save the following variables in `vpc_sc_project/terraform.tfvars`:
+        - `project_id`: create a *different* project ID from the one in step 1
+        - `service_account_file`
+        - `organization_id`: should be *same* as in step 1
+        - `billing_account_id`: *same* as in step 1
+        - `cloud_router_region`: *same* as in step 1
+        - `shared_secret_string_for_vpn_connection`: *same* as in step 1
+        - Leave the rest blank.
+    - From a command line terminal:
+        - `cd ../vpc_sc_project`
+        - `terraform plan`
+            - It should show a plan to create one project, one network, one subnet, one firewall rule, one route.
+        - `terraform apply`
+            - You might not get the "Compute Engine API" error message here, but if you do, enable it and try again.
+
+3. Reserve a public IP for your "on-prem" VPN router (to be clear, further use of "on-prem" in these instructions refers to the GCP project that we're pretending is your on-prem environment).
+    - Open `onprem_project/main.tf` in your favorite text editor.
+    - Search for `STEP 1A`.
+    - Don't actually delete the line; just add `*/` to the **end** of the line.
```

Comment:
Requiring commenting/uncommenting is a rather confusing workflow. Can you explain why we need this?

Reply:
I agree. There are a lot of manual steps to this guide that make it daunting to follow. Could we not use more Terraform configuration to obviate the need for file edits and orchestration? Variables and remote states come to mind.

Reply:
Here's a suggestion to take out all the commenting and uncommenting steps and embed some Terraform best practices in this demo.

Instead of a per project folder. Create folders per resource category. Here's an example folder structure:

examples/onprem_demo/projects (for all projects resources and to configure the [GCS backend](https://www.terraform.io/docs/backends/types/gcs.html)
examples/onprem_demo/network (for all networks, VPNs, firewall rules, routers, etc.. in both projects)
examples/onprem_demo/compute (for all VMs created in both projects)
examples/onprem_demo/vpc_sc ( for all VPC SC resources created in the VPC SC project)

Here's a [lab](https://github.com/Tfmenard/terraform-vpn-lab) I made from which you can reuse code

Reply:
We need the commenting/uncommenting because we need to first reserve public IP's for both the on-prem and VPC SC projects, for the VPN router; then we need to "tell" each environment what the other environment's VPN router's IP is.  So onprem project depends on the IP in VPC SC project, and VPC SC project depends on the IP in onprem project.

Reply:
You wouldn't need to do this if the VPNs in both projects were created in 1 root config/folder and you used implicit dependencies like I did [here](https://github.com/Tfmenard/terraform-vpn-lab/blob/6e613a31a6415fd5360192debe33b82c8e4c53e9/lab-vpn/network_vpn_gcp.tf#L49) in my lab.